### ISSUE TITLE
The dependencies are broken

### ISSUE THREAD

**Issue by chenyulue** [def-/ghc-datasize#10]
The dependencies listed in the `.cabal` file are broken, since `ghc-heap` package does not exit in the hackage. Actually, substituting `ghc-heap` with `ghc-lib-parser`  works. 

So please update the dependencies.

**Comment by def-** [comment]
Thanks for report, fixed.

**Comment by noughtmare** [comment]
For some reason the Hackage package doesn't have the same dependencies as the package on GitHub. I still get errors with version 0.2.3 on Hackage, but building from this GitHub repo works fine.

**Comment by def-** [comment]
Weird, I don't see why. What is the exact error?

**Comment by noughtmare** [comment]
```
$ cabal repl -b ghc-datasize
Resolving dependencies...
Build profile: -w ghc-8.10.4 -O1
In order, the following will be built (use -v for more details):
 - ghc-datasize-0.2.3 (lib) (requires build)
 - fake-package-0 (lib) (first run)
Starting     ghc-datasize-0.2.3 (lib)
Building     ghc-datasize-0.2.3 (lib)

Failed to build ghc-datasize-0.2.3.
Build log (
/home/jaro/.cabal/logs/ghc-8.10.4/ghc-datasize-0.2.3-f95cf5ca0c212b0b68a002dd914c8d0351ea882f0d0b604f1e168d81a08a8259.log
):
Configuring library for ghc-datasize-0.2.3..
Preprocessing library for ghc-datasize-0.2.3..
Building library for ghc-datasize-0.2.3..
[1 of 1] Compiling GHC.DataSize     ( src/GHC/DataSize.hs, dist/build/GHC/DataSize.o, dist/build/GHC/DataSize.dyn_o )

src/GHC/DataSize.hs:23:1: error:
    Could not load module ‘GHC.Exts.Heap’
    It is a member of the hidden package ‘ghc-lib-parser-9.0.1.20210207’.
    Perhaps you need to add ‘ghc-lib-parser’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘ghc-lib-parser-8.10.4.20210206’.
    Perhaps you need to add ‘ghc-lib-parser’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘ghc-heap-8.10.4’.
    Perhaps you need to add ‘ghc-heap’ to the build-depends in your .cabal file.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
23 | import GHC.Exts.Heap hiding (size)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/GHC/DataSize.hs:24:1: error:
    Could not load module ‘GHC.Exts.Heap.Constants’
    It is a member of the hidden package ‘ghc-lib-parser-9.0.1.20210207’.
    Perhaps you need to add ‘ghc-lib-parser’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘ghc-lib-parser-8.10.4.20210206’.
    Perhaps you need to add ‘ghc-lib-parser’ to the build-depends in your .cabal file.
    It is a member of the hidden package ‘ghc-heap-8.10.4’.
    Perhaps you need to add ‘ghc-heap’ to the build-depends in your .cabal file.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
24 | import GHC.Exts.Heap.Constants (wORD_SIZE)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cabal: Failed to build ghc-datasize-0.2.3 (which is required by
fake-package-0). See the build log above for details.
```

**Comment by noughtmare** [comment]
But also [on the ghc-datasize page on Hackage](http://hackage.haskell.org/package/ghc-datasize) you can see that it depends on `ghc-lib-parser-ex` without version bounds.

**Comment by def-** [comment]
Yes, I'm wondering how that happened since I uploaded it with the version bounds.

**Comment by collares** [comment]
Just to check: The problem reported by @noughtmare is that the Git version depends on `ghc-lib-parser`, while the Hackage version depends on `ghc-lib-parser-ex` and therefore [fails to build](https://hackage.haskell.org/package/ghc-datasize-0.2.3/reports/3). Am I understanding it correctly?

**Comment by noughtmare** [comment]
@collares, yes. The [.cabal file on hackage](https://hackage.haskell.org/package/ghc-datasize-0.2.3/src/ghc-datasize.cabal) is mysteriously different from the [.cabal file here in git](https://github.com/def-/ghc-datasize/blob/620ac987d39f5c93a6d93e9fa91127318d585e1e/ghc-datasize.cabal).

**Comment by def-** [comment]
Released again as 0.2.4, hope it works now: https://hackage.haskell.org/package/ghc-datasize-0.2.4/src/ghc-datasize.cabal
